plugins/blip/basics/logic/datamunger.js

```diff
@@ -174,10 +179,71 @@ module.exports = function(bgClasses) {
         averageDailyCarbs: sumCarbs/((Date.parse(end) - Date.parse(start))/constants.MS_IN_DAY)
       };
     },
-    infusionSiteHistory: function(basicsData) {
-      var infusionSitesPerDay = basicsData.data.reservoirChange.dataByDate;
+    getLatestPumpUploaded: function(patientData) {
+      var latestPump = _.last(patientData.grouped.pumpSettings);
+
+      if (latestPump && latestPump.hasOwnProperty('source')) {
+        return latestPump.source;
+      }
+
+      return null;
+    },
+    processInfusionSiteHistory: function(basicsData, latestPump, patient) {
+      if (!latestPump) {
+        return null;
+      }
+
+      var {
+        permissions,
+        profile: {
+          fullName,
+          patient: {
+            settings
+          },
+        },
+      } = patient;
+
+      var hasUploadPermission = permissions.hasOwnProperty('admin') || permissions.hasOwnProperty('root');
+
+      basicsData.sections.siteChanges.selectorMetaData = {
+        latestPump: latestPump,
+        canUpdateSettings: hasUploadPermission,
+        patientName: fullName,
+      };
+
+      if (latestPump === constants.ANIMAS || latestPump === constants.MEDTRONIC || latestPump === constants.TANDEM) {
+        basicsData.data.cannulaPrime.infusionSiteHistory = this.infusionSiteHistory(basicsData, constants.SITE_CHANGE_CANNULA);
+        basicsData.data.tubingPrime.infusionSiteHistory = this.infusionSiteHistory(basicsData, constants.SITE_CHANGE_TUBING);
+
+        if (settings && settings.hasOwnProperty('siteChangeSource')) {
+          basicsData.sections.siteChanges.type = settings.siteChangeSource;
+          basicsData.sections.siteChanges.selectorOptions = basicsActions.setSelected(basicsData.sections.siteChanges.selectorOptions, settings.siteChangeSource);
+        }
+        else {
+          basicsData.sections.siteChanges.type = constants.SECTION_TYPE_UNDECLARED;
+          basicsData.sections.siteChanges.settingsTogglable = togglableState.open;
+        }
+      }
+      else if (latestPump === constants.INSULET) {
+        basicsData.data.reservoirChange.infusionSiteHistory = this.infusionSiteHistory(basicsData, constants.SITE_CHANGE_RESERVOIR);
+
+        basicsData.sections.siteChanges.type = constants.SITE_CHANGE_RESERVOIR;
+        basicsData.sections.siteChanges.selector = null;
+        basicsData.sections.siteChanges.settingsTogglable = togglableState.off;
+      }
+      // Medtronic (CareLink) or other unsupported pump
+      else {
```

Comment:
I updated it. But to me `Medtronic (CareLink)` and `CareLink (Medtronic)` mean essentially the same thing. Is this just how Tidepool has historically referred to it and I'm not aware? My reason for listing as Medtronic (CareLink) is that direct Medtronic uploads are called Medtronic Direct, so I figured this would be a good way to specify that these were Medtronic uploaded through (CareLink).